index.js

```diff
@@ -137,4 +137,38 @@ newSightingForm.addEventListener('submit', function (event) {
 /***** End of Starter Code *****/
 /************************** EVENTS PART 2 JS MINI CHALLENGE ******************************/
 
+// Deliverable 1 + 2 : Like an animal sighting + Delete an animal sighting
+// When a user clicks an animal sighting's like button, that animal sighting's likes should increase by 1.
+// When a user clicks an animal sighting's delete button, that animal sighting should be removed from the page. 
+
+
+const animalsUl = document.querySelector('ul#animals')
+
+animalsUl.addEventListener('click', function(event) {
+
+    if (event.target.matches('button.like-button')) {
+        const pLikesDisplay = event.target.previousElementSibling 
+        const likes = parseInt(pLikesDisplay.textContent) + 1
+        pLikesDisplay.textContent = `${likes} Likes`
+    } 
+    else if (event.target.matches('button.delete-button')) {
+        const li = event.target.closest('li')
+        li.remove()
+    } 
+    else if (event.target.matches('button.toggle-update-form-button')) {
+        const updateForm = event.target.nextElementSibling
+        if (updateForm.style.display === 'none') {
+            updateForm.style.display = 'block'
+        } else {
+            updateForm.style.display = 'none'
+        }
+    }
+})
```

Comment:
Nice! You have an event listener on the entire animals UL element, and based on the class names you are delegating out different logic. Great. First step is getting everything up and running. Awesome. Next, we can refactor for readability and to make our code more reusable.

```
/***** DELIVERABLE 1, 2, & 3 part 1 *****/

const animalsUL = document.querySelector('ul#animals')
animalsUL.addEventListener("click", function(event){
    if (event.target.matches('button.like-button')){
        handleLike(event)
    } else if (event.target.matches('button.delete-button')){
        handleDelete(event)
    } else if (event.target.matches('button.toggle-update-form-button')){
        handleToggleUpdate(event)
    }
})

/***** DELIVERABLE 1 part 2 *****/
function handleLike(e){
    const likesDisplayPTag = e.target.previousElementSibling
    const currentTextContent = likesDisplayPTag.innerText
    let numberOfLikes = parseInt(currentTextContent.split(" ")[0])
    numberOfLikes += 1
    const newTextContent = `${numberOfLikes} Likes`
    likesDisplayPTag.innerText = newTextContent
}

/***** DELIVERABLE 2 part 2 *****/
function handleDelete(e){
    const parentLI = e.target.parentElement
    parentLI.remove()
}

/***** DELIVERABLE 3 part 2 *****/
function handleToggleUpdate(e){
    const updateForm = e.target.nextElementSibling
    toggleHidden(updateForm)
}

/***** DELIVERABLE 3 part 3 *****/
function toggleHidden(element){
    element.style.display === "none" ? element.style.display = "block" : element.style.display = "none"
}
```